src/overlays/actors/ovl_Bg_Bombwall/z_bg_bombwall.c

```diff
@@ -36,29 +61,192 @@ const ActorInit Bg_Bombwall_InitVars = {
+
+void BgBombwall_InitDynapoly(BgBombwall* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    ColHeader* colHeader;
+
+    colHeader = NULL;
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_050041B0, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf((const char*)"Warning : move BG 登録失敗(%s %d)(arg_data 0x%04x)\n", "../z_bg_bombwall.c", 0xF3,
+                     this->dyna.actor.params);
+    }
+}
+
+void BgBombwall_RotateVec(Vec3f* arg0, Vec3f* arg1, f32 arg2, f32 arg3) {
+    arg0->x = (arg1->z * arg2) + (arg1->x * arg3);
+    arg0->y = arg1->y;
+    arg0->z = (arg1->z * arg3) - (arg1->x * arg2);
+}
+
+void BgBombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+
+    s32 i;
+    s32 j;
+    Vec3f vecs[3];
+    Vec3f sp80;
+
+    s32 pad;
+    BgBombwall* this = THIS;
+
+    f32 sin = Math_Sins(this->dyna.actor.shape.rot.y);
+    f32 cos = Math_Coss(this->dyna.actor.shape.rot.y);
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    Actor_SetScale(thisx, 0.1f);
+
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
+        func_8086EE94(this, globalCtx);
+    } else {
+
+        BgBombwall_InitDynapoly(this, globalCtx);
+        this->unk_2A2 |= 2;
+        Collider_InitTris(globalCtx, &this->collider);
+        Collider_SetTris(globalCtx, &this->collider, this, &sTrisInit, &(this->colliderItems[0]));
+
+        for (i = 0; i <= 2; i++) {
+
+            for (j = 0; j <= 2; j++) {
+
+                sp80.x = sTrisInit.list[i].dim.vtx[j].x;
+                sp80.y = sTrisInit.list[i].dim.vtx[j].y;
+                sp80.z = sTrisInit.list[i].dim.vtx[j].z + 2.0f;
+
+                BgBombwall_RotateVec(&vecs[j], &sp80, sin, cos);
+
+                vecs[j].x += this->dyna.actor.posRot.pos.x;
+                vecs[j].y += this->dyna.actor.posRot.pos.y;
+                vecs[j].z += this->dyna.actor.posRot.pos.z;
+            }
+            func_800627A0(&this->collider, i, &vecs[0], &vecs[1], &vecs[2]);
+        }
+        this->unk_2A2 |= 1;
+
+        BgBombwall_setFuncPointers(this, globalCtx);
+    }
+    osSyncPrintf("(field keep 汎用爆弾壁)(arg_data 0x%04x)(angY %d)\n", this->dyna.actor.params,
+                 this->dyna.actor.shape.rot.y);
+}
+
+void BgBombwall_DestroyInternal(BgBombwall* this, GlobalContext* globalCtx) {
+    if (this->unk_2A2 & 2) {
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->unk_2A2 &= 0xfffd;
+    }
+    if (this->unk_2A2 & 1) {
+        Collider_DestroyTris(globalCtx, &this->collider);
+        this->unk_2A2 &= 0xfffe;
+    }
+}
+
+void BgBombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgBombwall_DestroyInternal((BgBombwall*)thisx, globalCtx);
```

Comment:
you should be able to make a temp for this using the THIS macro and pass that instead of casting